### PR TITLE
Changed inventory vulnerabilities references to links

### PR DESCRIPTION
**Pull request by JuanGarriuz**
### Description

This pull converts Vulnerability Detector reference fields into links.
Additionally, the vulnerability injector script is modified to occasionally add 2 links.
 
### Issues Resolved

- https://github.com/wazuh/wazuh-dashboard-plugins/issues/6784

### Evidence

image

### Test

Go to Vulnerability Detection > Inventory, and select vulnerability.reference columns and check links are functional

### Check List
- [ ] All tests pass
  - [ ] `yarn test:jest`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by Machi3mfl — APPROVED** [review]
CR: ✅ 
Test: ✅ 

Screenshot 2024-09-04 at 11 37 32

https://github.com/user-attachments/assets/dcdf7676-fb5f-4b9a-8c0d-98d272d34d75

**Review by yenienserrano — DISMISSED** [review]
LGTM

image

File: plugins/main/public/components/common/wazuh-discover/render-columns.tsx [hunk 1]
```diff
@@ -79,6 +79,29 @@ export const wzDiscoverRenderColumns: tDataGridRenderColumn[] = [
...
+          {value?.split(',').map((link, index) => (
+            <span key={index}>
+              {!!index && ', '}
+              <EuiToolTip
+                position='top'
+                content='Navigate to the vulnerability reference'
+              >
+                <EuiLink href={link} target='_blank'>
+                  {link}
+                </EuiLink>
```
Comment (Desvelao) [inline plugins/main/public/components/common/wazuh-discover/render-columns.tsx:96]: **nitpick**: the external links should have `rel` and `external` properties:

```tsx
<EuiLink
  href={}
  target='_blank'
  rel='noopener noreferrer'
  external
>
  View surrounding documents
</EuiLink>

```

**Review by yenienserrano — DISMISSED** [review]

**Review by Machi3mfl — DISMISSED** [review]

**Review by guidomodarelli — APPROVED** [review]
:green_circle: CR
:green_circle: Test

image

**Comment by Desvelao** [comment]
# Test

image
image

Expanded cell
image

File: plugins/main/public/components/common/wazuh-discover/render-columns.tsx [hunk 2]
```diff
@@ -79,6 +79,34 @@ export const wzDiscoverRenderColumns: tDataGridRenderColumn[] = [
       );
     },
   },
+  {
+    id: 'vulnerability.reference',
+    render: value => {
+      const links = (
+        <>
+          {value?.split(',').map((link, index) => (
```
Comment (Desvelao) [inline plugins/main/public/components/common/wazuh-discover/render-columns.tsx:87]: **nitpick**: the `vulnerability.reference` value seems to be an string that contains the URLs separated by `, ` (comma-whitespace). So we should transform the string to an array splitting by the same separator, `, ` (comma-whitespace)

**Comment by JuanGarriuz** [comment]
image

Added to the vulnerability injector script multiple links

**Review by chantal-kelm — APPROVED** [review]
image

**Pull request merged**